### PR TITLE
Add fluid spacing scale

### PR DESCRIPTION
**Pull request by richtabor**
**Description**

Add a consistent fluid spacing scale that we can start leveraging in patterns, instead of using arbitrary values. This proposal uses the Fibonacci sequence. 

We can continue to iterate and refine the values, but I think it'd be good to have a consistent set to start working from. 

**Testing Instructions**

1. Activate the theme. 
2. Add spacer block. 
3. Adjust the height to see the spacing presets. 
4. View on mobile to see the fluidity. 

**Screenshots**

Desktop: 
CleanShot 2023-08-27 at 10 56 52

Mobile: 
CleanShot 2023-08-27 at 11 04 14

Example of this fluid spacing expressed in a pattern, with `40` as the outer padding: 
CleanShot 2023-08-28 at 09 44 47
CleanShot 2023-08-28 at 09 44 58

### CODE REVIEW

File: theme.json [hunk 1]
```diff
@@ -115,13 +115,38 @@
...
+					"slug": "10"
+				},
+				{
+					"name": "2",
+					"size": "min(36px, 6vw)",
+					"slug": "20"
+				},
+				{
+					"name": "3",
+					"size": "min(60px, 10vw)",
```
Comment (MaggieCabrera) [inline theme.json]: Should we use rems instead of px here?

Comment (richtabor) [reply theme.json]: Good call, incoming.

Comment (richtabor) [reply theme.json]: Done. Thanks!

**Review by MaggieCabrera — APPROVED** [review]
Love this, thank you!

**Comment by beafialho** [comment]
These presets look good!

I'm seeing a lot of spacing that was not supposed to be there, particularly in stack blocks in the home template. This may have happened everywhere else. It seems that it may have inheriting one of the presets, but one that's way too high? Here's what I mean, comparing it to a demo site I created with the intended spacings:

https://github.com/WordPress/twentytwentyfour/assets/65220155/097cc565-5fec-4b68-8079-3f893cc863e1

**Comment by richtabor** [comment]
Yea, we need to adjust the templates to use the system.

**Pull request merged**